### ISSUE TITLE
warp_image_tps: default align_corners=False breaks even identity warps (create_meshgrid hardcodes the True convention)

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#3928]
Found during the #3926 convention audit (execution-verified, torch 2.9.1):

`warp_image_tps` defaults to `align_corners=False`, but its internal grid comes from `create_meshgrid(..., normalized_coordinates=True)`, which maps pixel 0 → −1 and pixel W−1 → +1 — the `align_corners=True` convention. The default therefore samples with a mismatched grid and even a mathematically-identity TPS transform corrupts the image:

```python
img = torch.arange(16.0).view(1, 1, 4, 4)
src = torch.tensor([[[-1., -1.], [-1., 1.], [1., -1.], [1., 1.], [0., 0.]]])
kernel, affine = get_tps_transform(src, src)
out = warp_image_tps(img, src, kernel, affine)           # default: max |out - img| ≈ 0.6
out = warp_image_tps(img, src, kernel, affine, align_corners=True)  # ≈ 2e-7
```

Candidate fixes: default flip (breaking, needs deprecation), or thread `align_corners` into the meshgrid normalization so both settings self-consist. Documented as a `.. warning::` in #3926 pending this fix. Split out of the closed umbrella #3923 per review feedback.

—
*Issue authored by Claude (Claude Code) on behalf of @ducha-aiki.*
